Always create serdag in dagmaker fixture

dstandish opened the pull request:
This PR makes it so dag_maker always creates serialized dag and dagversion objects, along with dag model.  

It's not really sensible anymore to not have these other objects there; they are always there in production, and increasingly we need them there for code to work right, and their omission can leave some bugs hidden (and some of them are resolved as part of this).

Initially, I was going to just remove the option, but, it _also_ controls the type of dag object returned by dag_maker (serdag vs dag), so for now I leave that as is.

dstandish committed:
Always create serdag in dagmaker fixture

ephraimbuddy reviewed (commented):
I worked on making it backwards compatible here: https://github.com/apache/airflow/pull/46703/files#diff-b666e1a99da023285548b78cbdf107890577203173a6ab31fb0722491ed1b280. 

You might want to take a look

dstandish committed:
Fix tests

dstandish committed:
handle rebase conflicts

dstandish committed:
tests

dstandish committed:
mark two tests xfail

dstandish committed:
fix test

dstandish committed:
fix docker test

dstandish committed:
fix test

dstandish committed:
fix tests

uranusjr committed:
Make asset event trigger serialization more robust

The serializaed trigger must be JSON-compatible to be stored in the
database, so we may need to run it through BaseSerialization.

I added some extra code so the serialization is done at a minimum, and
not shown to the user if possible.

dstandish committed:
fix test

uranusjr commented on airflow-core/tests/unit/api_fastapi/conftest.py:
```diff
@@ -151,17 +150,20 @@ def make_dag_with_multiple_versions(dag_maker, configure_git_connection_for_dag_
-        SerializedDagModel.write_dag(
-            dag, bundle_name="dag_maker", bundle_version=f"some_commit_hash{version_number}"
-        )
+        session.commit()
```
Do we have to commit twice every iteration?

dstandish replied:
removed

uranusjr commented on airflow-core/tests/unit/dag_processing/test_collection.py:
```diff
@@ -121,6 +121,7 @@ def per_test(self) -> Generator:
         yield
         self.clean_db()
 
+    @pytest.mark.xfail(reason="test fails on serialization of the watcher trigger kwargs")
```
I think this can now be removed

dstandish replied:
doen

uranusjr commented on airflow-core/tests/unit/dag_processing/test_collection.py:
```diff
@@ -287,6 +288,7 @@ def test_add_asset_activate_already_exists(self, dag_maker, session):
         session.flush()
         assert orm_assets["myasset", "file://myasset/"].active is not None, "should pick up existing active"
 
+    @pytest.mark.xfail(reason="conflict not handled when serdag creation enabled")
```
Is there an issue for this?

dstandish replied:
yeah it was this one https://github.com/apache/airflow/issues/50405

i can now remove the xfail here also

uranusjr commented on airflow-core/tests/unit/models/test_taskinstance.py:
```diff
@@ -4719,17 +4719,18 @@ def test_map_has_dag_version(self, dag_maker, session):
+        # get the dag version for the dag
+        dag_version = session.scalar(select(DagVersion).where(DagVersion.dag_id == dag.dag_id))
+        # dag_version = session.merge(DagVersion(dag_id="test", bundle_name="test"))
+        # session.commit()
```
```suggestion
```

dstandish committed:
Update airflow-core/tests/unit/models/test_taskinstance.py

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

dstandish committed:
remove some xfails and a commit

kaxil commented on airflow-core/src/airflow/serialization/serialized_objects.py:
```diff
@@ -230,12 +230,26 @@ def __str__(self) -> str:
 
 
 def _encode_trigger(trigger: BaseEventTrigger | dict):
```
The changes in this PR potentially caused regression: https://github.com/apache/airflow/issues/51809


Mis-leading PR title
